wdl/transforms/draft3/src/test/scala/AstToWorkflowDefinitionElementSpec.scala

```diff
@@ -0,0 +1,97 @@
+import wom.types.{WomSingleFileType, WomStringType}
+import wdl.transforms.base.ast2wdlom.AstToWorkflowDefinitionElement._
+
+
+class AstToWorkflowDefinitionElementSpec extends AnyFlatSpec{
+  behavior of "Check Stdouts and Stderrs"
+
+  val mockInputSectionStdout = Option(InputsSectionElement(Vector(InputDeclarationElement(PrimitiveTypeElement(WomSingleFileType), "i", Some(StdoutElement)))))
+  val mockInputSectionStderr = Option(InputsSectionElement(Vector(InputDeclarationElement(PrimitiveTypeElement(WomSingleFileType), "i", Some(StderrElement)))))
+  val mockInputSectionNonStd = Option(InputsSectionElement(Vector(InputDeclarationElement(PrimitiveTypeElement(WomStringType), "more", Some(StringLiteral("more"))))))
+
+
+  val mockIntermediatesStdout = Vector(IntermediateValueDeclarationElement(PrimitiveTypeElement(WomSingleFileType), "y", StdoutElement))
+  val mockIntermediatesStderr = Vector(IntermediateValueDeclarationElement(PrimitiveTypeElement(WomSingleFileType), "y", StderrElement))
+  val mockIntermediatesNonStd = Vector(IntermediateValueDeclarationElement(PrimitiveTypeElement(WomStringType), "here", StringLiteral("here")))
+
+  val mockOutputSectionStdout = Option(OutputsSectionElement(Vector(OutputDeclarationElement(PrimitiveTypeElement(WomSingleFileType), "s", StdoutElement))))
+  val mockOutputSectionStderr = Option(OutputsSectionElement(Vector(OutputDeclarationElement(PrimitiveTypeElement(WomSingleFileType), "s", StderrElement))))
+  val mockOutputSectionNonStd = Option(OutputsSectionElement(Vector(OutputDeclarationElement(PrimitiveTypeElement(WomStringType), "more", StringLiteral("more")))))
+
+
+  it should "return an error when there is an stdout in input section" in {
+    val testInputs = checkDisallowedInputElement(mockInputSectionStdout, StdoutElement, "stdout")
+    testInputs match {
+      case Valid(_) => fail("Input section contained stdout. Should have failed.")
+      case Invalid(e) => e.head should be("Workflow cannot have stdout expression in input section at workflow-level.")
+    }
+  }
+
+  it should "return an error when there is an stderr in input section" in {
+    val testInputs = checkDisallowedInputElement(mockInputSectionStderr, StderrElement, "stderr")
+    testInputs match {
+      case Valid(_) => fail("Input section contained stderr. Should have failed.")
+      case Invalid(e) => e.head should be("Workflow cannot have stderr expression in input section at workflow-level.")
+    }
+  }
+
+  it should "not return an error for non-stdout/stderr in the inputs section" in {
+    val testInputs = checkDisallowedInputElement(mockInputSectionNonStd, StdoutElement, "non-stdout/stderr")
+    testInputs match {
+      case Valid(_) => "Input section contained a non-stdout/stderr element."
+      case Invalid(e) => e.head should be("Workflow failed at input section.")
+    }
+  }
```

Comment:
For this (and below similar) test case where we have non-stdout/stderr in the inputs section, we want to fail the test if the `checkDisallowedInputElement` returned `Invalid`. So here the in the `Invalid` case you would need to return `fail("Check shouldn't have returned error as input section had non-stdout/stderr inputs")`. And for the `Valid` case you can simply return nothing: `// No action`